netadapter/netadapter_test.go

```diff
@@ -0,0 +1,150 @@
+package netadapter
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kaspanet/kaspad/domainmessage"
+
+	"github.com/kaspanet/kaspad/config"
+	"github.com/kaspanet/kaspad/netadapter/router"
+)
+
+// getRouterInitializer returns new RouterInitializer which simply sets
+// new incoming route for router and stores this route in map for further usage in tests
+func getRouterInitializer(routes *sync.Map, routeName string,
+	t *testing.T, wg *sync.WaitGroup) func(*router.Router, *NetConnection) {
+	return func(router *router.Router, connection *NetConnection) {
+		route, err := router.AddIncomingRoute([]domainmessage.MessageCommand{domainmessage.CmdPing})
+		if err != nil {
+			t.Fatalf("TestNetAdapter: AddIncomingRoute failed: %+v", err)
+		}
+		routes.Store(routeName, route)
+		wg.Done()
+	}
+}
+
+func TestNetAdapter(t *testing.T) {
+	timeout := time.Second * 5
```

Comment:
make timeout, nonce, host, ports constants. (You can still keep them inline, no need to move outside function)